### PR TITLE
[s] Clowncar exploit fix

### LINKED ISSUE #43233: Clown Car Can Yoink You From Across The Map
Clown cars can suck adjacent people into themselves by click+dragging their victim into/onto the clown car, which starts a short timer that will suck the victim into the clown car when it completes. That's fine. The problem comes in that the victim only needs to be adjacent to the clown car when it BEGINS to stuff them inside of the car, and there are no checks to see if the victim is still adjacent when it FINISHES stuffing them into the car.

What this means is that, when the stuffing timer finishes, no matter where you are in relation to the clown car, you'll get sucked into it.

An example:
Mr. Honkerton is cruising around in his clown car and sees Secman McGee, who is crawling to avoid being sucked up by the clown car. Mr. Honkerton swoops into a spot that's adjacent to Secman and click+drags Secman onto his clown car, beginning the stuffing process. Secman notices this, gets up, and runs as fast as he can away from the clown car while high on every speed chem in the game, getting about 2 screens away from the clown car before the stuffing process completes. Unfortunately, Secman's efforts were futile, as he was sucked into the clown car when the stuffing process finished despite not even being within visual range of the clown car.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #43233
:cl: ShizCalev
fix: Fixed an exploit allowing you to grab people from anywhere with a clowncar.
/:cl:

### CODE REVIEW

**Comment by fluffe9911** [comment]
clown car should prob be able to grab people who rest now honestly due to the fact resting people can now crawl around and fire weapons at the clown car and the clown car cant do anything to fight back but nice fix

**Comment by ShizCalev** [comment]
@fluffe9911 i'd lean more towards having them be hurt like mulebots do when they run you over, that'd be for a different PR though.

**Comment by ATH1909** [comment]
> @fluffe9911 i'd lean more towards having them be hurt like mulebots do when they run you over, that'd be for a different PR though.

That's actually a really good idea.

Also, thanks for making a PR to fix this so quickly.

File: code/modules/vehicles/cars/car.dm [hunk 1]
```diff
@@ -78,11 +78,16 @@
...
 		return FALSE
-	if(do_mob(forcer, get_enter_delay(M), target = src))
+	var/atom/old_loc = loc
+	if(do_mob(forcer, M, get_enter_delay(M), extra_checks=CALLBACK(src, /obj/vehicle/sealed/car/proc/is_car_stationary, old_loc)))
 		mob_forced_enter(M, silent)
 		return TRUE
 	return FALSE
 
+/obj/vehicle/sealed/car/proc/is_car_stationary(atom/old_loc)
+	if(old_loc == loc)
```
Comment (alexkar598) [inline code/modules/vehicles/cars/car.dm]: you could do `return (old_loc == loc)`

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**